impeller/geometry/rect.h

```diff
@@ -190,15 +207,44 @@ struct TRect {
     return {GetRight(), GetBottom()};
   }
 
+  /// @brief  Get the area of the rectangle, equivalent to |GetSize().Area()|
+  constexpr T Area() const { return size.Area(); }
+
   /// @brief  Get the center point as a |Point|.
   constexpr Point GetCenter() const {
     return Point(origin.x + size.width * 0.5f, origin.y + size.height * 0.5f);
   }
 
+  /// @brief  Get the point related to this rectangle associated with
+  ///         the given fractional values |tx| and |ty| with a fractional
+  ///         value of 0 referring to the left or upper edge of the
+  ///         rectangle and a fractional value of 1 referring to the right
+  ///         or bottom edge of the rectangle.
+  ///
+  ///         GetRelative(0, 0) returns the upper left corner
+  ///         GetRelative(0.5, 0.5) returns the center of the rectangle
+  ///         GetRelative(1, 1) returns the lower right corner
+  constexpr Point GetRelative(Scalar tx, Scalar ty) const {
+    return {origin.x + size.width * tx, origin.y + size.height * ty};
+  }
+
+  /// @brief  Get the pointe related to this rectangle associated with
+  ///         the fractional values in the point |t|, equivalent to
+  ///         |GetRelative(t.x, t.y)|
+  constexpr Point GetRelative(const Point& t) const {
+    return GetRelative(t.x, t.y);
+  }
```

Comment:
When reading the code using these methods it wasn't immediately apparent to me what they actually do. 

I wonder if there's some clearer name we could use? Looking at the Dart Rect and SkRect classes they don't quite have a convenience for this (though the Dart API has things like `rect.topCenter`).

I also wonder if we should DCHECK that `0 <= t <= 1`, although maybe that can't be constexpr anymore...

Reply:
I didn't see a need for the [0,1] restriction, though. What if someone wants to find the point that is halfway to the left of the rectangle?

I agree the name isn't great and would love to see some suggestions.

Reply:
Another thought - the code that uses this was only using 0,1 values and doing it to choose which corner was being inserted in a triangle list. That code could be rewritten to use a triangle strip and getPoints (which I believe is ordered for making a triangle strip) and we could avoid this new method entirely. But, I'd have to make sure that change was tested.

The code is in atlas_contents.cc seen modified in this PR...

Reply:
New methods deleted in favor of a simpler implementation in atlas_contents.cc using existing methods.